packages/connect/src/utils/debug.ts

```diff
@@ -109,3 +153,13 @@ export const getLog = () => {
     logs.sort((a, b) => a.timestamp - b.timestamp);
     return logs;
 };
+
+export const initSharedLogger = (workerSrc: string) => {
+    const worker = new SharedWorker(workerSrc);
+    worker.port.start();
+    const logWriterFactory = (): LogWriter => ({
+        add: (message: LogMessage) =>
+            worker.port.postMessage({ type: 'add-log', data: JSON.parse(JSON.stringify(message)) }),
```

Comment:
and hopefully the very last requirement - this JSON.parse JSON stringify is maybe not necessary anymore. If we send some huge object it would just eat resources without us noticing it. We already have a catch block on line 70 so should there be some forgotten cyclical reference, we would learn it there

Reply:
Ups, yes that should not be necessary. Good catch!

Reply:
hm but it looks like we are actually sending something that can't be serialized by the internal communication mechanisms 
<img width="451" alt="image" src="https://github.com/trezor/trezor-suite/assets/30367552/f7be74d2-064f-4809-80d6-4787dac2979b">

Reply:
also please test it in firefox! 
this one particulary indicates that we are trying to log something that shoudn't be logged at all

Reply:
Fixed it by https://github.com/trezor/trezor-suite/pull/8623/commits/5a19241b73b54df776042eb91aa1e45c5d8c937a

And added a test that will check the logs in console and fail if there are any. For now I have to add one exception because we have stale error because of favicon, that PR https://github.com/trezor/trezor-suite/pull/9377 is trying to fix. So after the other PR is merged, it should not have exceptions.

https://github.com/trezor/trezor-suite/pull/8623/files#diff-1724a9e2ce31b052b48025b50cb0660c52f6ed77635ab2d3a57dbec7fcab50f4R81-R83